x-pack/legacy/plugins/reporting/export_types/csv/server/create_job.ts

```diff
@@ -4,27 +4,17 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { ReportingCore } from '../../../server';
 import { cryptoFactory } from '../../../server/lib/crypto';
-import {
-  ConditionalHeaders,
-  CreateJobFactory,
-  ESQueueCreateJobFn,
-  RequestFacade,
-} from '../../../types';
+import { ScheduleTaskFnFactory, TaskScheduleFn } from '../../../types';
 import { JobParamsDiscoverCsv } from '../types';
 
-export const createJobFactory: CreateJobFactory<ESQueueCreateJobFn<
+export const createJobFactory: ScheduleTaskFnFactory<TaskScheduleFn<
   JobParamsDiscoverCsv
->> = function createJobFactoryFn(reporting: ReportingCore) {
+>> = function(reporting) {
   const config = reporting.getConfig();
   const crypto = cryptoFactory(config.get('encryptionKey'));
 
-  return async function createJob(
-    jobParams: JobParamsDiscoverCsv,
-    headers: ConditionalHeaders['headers'],
-    request: RequestFacade
-  ) {
```

Comment:
The types here are inferred, since `const createJobFactory` is initialized with a type